### PR TITLE
game win refactor

### PR DESCRIPTION
**Pull request by adamrysyk**
refactor game winning functions and add js docs

### CODE REVIEW

File: src/reducers/index.js [hunk 1]
```diff
@@ -200,8 +198,13 @@ const WINNING_LINES = [
...
 ];
 
-// logic from official react tutorial
-// https://reactjs.org/tutorial/tutorial.html
+/**
+ * based on logic from official react tutorial
+ * https://reactjs.org/tutorial/tutorial.html
+ * @param {Object[]} squares
+ * @returns {{ winningSymbol: string, winningPattern: number[] } | undefined}
+ * returning undefined means there is no winner
```
Comment (phi-tran-vibe) [inline src/reducers/index.js]: Tab
Also the word "returning" is redundant

File: src/reducers/index.js [hunk 2]
```diff
@@ -180,7 +179,6 @@ export default (state = initialState, action) => {
             return {
                 ...state,
                 matchOver,
-                gameOver,
```
Comment (phi-tran-vibe) [inline src/reducers/index.js]: why was this removed?

File: src/reducers/index.js [hunk 3]
```diff
@@ -212,37 +215,52 @@ function calculateMatchWinner(squares) {
...
-        winningPattern: undefined
-    };
 }
 
-function calculateGameWinner(state, categories, matchWinner, matchDraw) {
-    if (state.matchesToWin === state.categories.length) {
-        return calculateAllMatchesWon(categories, matchWinner, matchDraw);
+/**
+ * Determines which game winner function to call based on extra game options
+ * @param {number} matchesToWin - number of match wins needed to win the game
```
Comment (phi-tran-vibe) [inline src/reducers/index.js]: add another line above this
```
@param {Object} params
```
Then prefix all the param names with `params.`, Ex. `params.matchesToWin`, `params.categories`

This is because the function takes in 1 object as an argument
The way it's currently documented makes it seem like it takes in 4 separate arguments

File: src/reducers/index.js [hunk 4]
```diff
@@ -212,37 +215,52 @@ function calculateMatchWinner(squares) {
...
+        return calculateAllMatchesWon({ categories, matchWinner, matchDraw });
     }
-    return calculateMatchesToWinMet(state, categories);
+    return calculateMatchesToWinMet({ matchesToWin, categories });
 
 }
 
-function calculateAllMatchesWon(categories, matchWinner, matchDraw) {
+/**
+ * Calculate whether the player has won every match
```
Comment (phi-tran-vibe) [inline src/reducers/index.js]: description not needed, the function name is good enough

File: src/reducers/index.js [hunk 5]
```diff
@@ -212,37 +215,52 @@ function calculateMatchWinner(squares) {
...
+ * @param {Object[]} categories
```
Comment (phi-tran-vibe) [inline src/reducers/index.js]: same here

File: src/reducers/index.js [hunk 6]
```diff
@@ -212,37 +215,52 @@ function calculateMatchWinner(squares) {
...
     } else if (matchWinner === INCORRECT_SYMBOL || matchDraw) {
         return false;
     }
-    return undefined;
 }
 
-function calculateMatchesToWinMet(state, categories) {
+/**
+ * At the end of the game calculate if the player has won matchesToWin or more
+ * @param {number} matchesToWin - number of match wins needed to win the game
```
Comment (phi-tran-vibe) [inline src/reducers/index.js]: same here

File: src/reducers/index.js (hunk 1 above)
Comment (adamrysyk) [reply src/reducers/index.js]: done

File: src/reducers/index.js (hunk 2 above)
Comment (adamrysyk) [reply src/reducers/index.js]: not needed in state anymore - could potentially add it back and replace activeScreen: GAME_OVER

File: src/reducers/index.js (hunk 3 above)
Comment (adamrysyk) [reply src/reducers/index.js]: done

File: src/reducers/index.js (hunk 4 above)
Comment (adamrysyk) [reply src/reducers/index.js]: done

File: src/reducers/index.js (hunk 5 above)
Comment (adamrysyk) [reply src/reducers/index.js]: done

File: src/reducers/index.js (hunk 6 above)
Comment (adamrysyk) [reply src/reducers/index.js]: done
